test/tstDistributedTree.cpp

```diff
@@ -813,3 +813,106 @@ BOOST_AUTO_TEST_CASE_TEMPLATE(boost_comparison, DeviceType, ARBORX_DEVICE_TYPES)
   ARBORX_TEST_QUERY_TREE(ExecutionSpace{}, distributed_tree, within_queries,
                          query(ExecutionSpace{}, rtree, within_queries_host));
 }
+
+template <typename MemorySpace>
+class RayNearestPredicate
+{
+public:
+  RayNearestPredicate(
+      Kokkos::View<ArborX::Experimental::Ray *, MemorySpace> const &rays)
+      : _rays(rays)
+  {}
+
+  KOKKOS_FUNCTION std::size_t size() const { return _rays.extent(0); }
+
+  KOKKOS_FUNCTION ArborX::Experimental::Ray const &get(unsigned int i) const
+  {
+    return _rays(i);
+  }
+
+private:
+  Kokkos::View<ArborX::Experimental::Ray *, MemorySpace> _rays;
+};
+
+namespace ArborX
+{
+template <typename MemorySpace>
+struct AccessTraits<RayNearestPredicate<MemorySpace>, PredicatesTag>
```

Comment:
Just use
```suggestion
struct ArborX::AccessTraits<RayNearestPredicate<MemorySpace>, PredicatesTag>
```
without namespace scope.